scipy/spatial/qhull.pyx

```diff
@@ -2410,3 +2449,269 @@ class Voronoi(_QhullUser):
             self._ridge_dict = dict(zip(map(tuple, self.ridge_points.tolist()),
                                         self.ridge_vertices))
         return self._ridge_dict
+
+#------------------------------------------------------------------------------
+# Halfspace intersection interface, for Python
+#------------------------------------------------------------------------------
+
+class _QhalfUser(object):
```

Comment:
I pushed the desired changes. However, this introduces the nasty side-effect that the documentation for add_points and add_halfspaces no longer appears in their respective classes. I have no idea on how to display the hidden method's docstring there. I tried setting:
```python
self.add_points = self._QhullUser._add_points
```
To apparently no avail.